sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/_ownership_manager.py

```diff
@@ -23,74 +21,79 @@ class OwnershipManager(object):
 
     """
     def __init__(
-            self, eventhub_client: EventHubClient, consumer_group_name: str, owner_id: str,
-            partition_manager: PartitionManager, ownership_timeout: float
+            self, eventhub_client, consumer_group_name: str, owner_id: str,
+            partition_manager: PartitionManager, ownership_timeout: float,
+            partition_id: str
     ):
         self.cached_parition_ids = []  # type: List[str]
         self.eventhub_client = eventhub_client
+        self.fully_qualified_namespace = eventhub_client._address.hostname  # pylint: disable=protected-access
         self.eventhub_name = eventhub_client.eh_name
         self.consumer_group_name = consumer_group_name
         self.owner_id = owner_id
         self.partition_manager = partition_manager
         self.ownership_timeout = ownership_timeout
+        self.partition_id = partition_id
+        self._initializing = True
 
     async def claim_ownership(self):
         """Claims ownership for this EventProcessor
-        1. Retrieves all partition ids of an event hub from azure event hub service
-        2. Retrieves current ownership list via this EventProcessor's PartitionManager.
-        3. Balances number of ownership. Refer to _balance_ownership() for details.
-        4. Claims the ownership for the balanced number of partitions.
-
-        :return: List[Dict[Any]]
         """
         if not self.cached_parition_ids:
             await self._retrieve_partition_ids()
```

Comment:
Let's avoid having methods with side effects that the caller have to verify before calling the method.

Reply:
Not blocking for preview5, solving